File: src/main/java/com/likelionknu/applyserver/application/data/dto/response/ApplicationDetailResponse.java
package com.likelionknu.applyserver.application.data.dto.response;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@AllArgsConstructor
public class ApplicationDetailResponse {
    private Long applicationId;
    private String recruitTitle;
    private String status;
    private LocalDateTime startAt;
    private LocalDateTime endAt;
    private LocalDateTime submittedAt;
    private List<ApplicationAnswerResponse> answers;

    @Getter
    @AllArgsConstructor
    public static class ApplicationAnswerResponse {
        private Long questionId;
        private String question;
        private String answer;
    }
}

File: src/main/kotlin/com/likelionknu/applyserver/application/data/dto/response/ApplicationDetailResponse.kt
package com.likelionknu.applyserver.application.data.dto.response

import java.time.LocalDateTime

data class ApplicationDetailResponse(
    val applicationId: Long,
    val recruitId: Long,
    val recruitTitle: String,
    val status: String,
    val startAt: LocalDateTime,
    val endAt: LocalDateTime,
    val submittedAt: LocalDateTime?,
    val answers: List<ApplicationAnswerResponse>
) {
    data class ApplicationAnswerResponse(
        val questionId: Long,
        val question: String,
        val answer: String
    )
}